### PR TITLE
Add dask-jobqueue and dask-drmaa links to HPC setup docs

### PR DESCRIPTION
**Pull request by mrocklin**
cc @jakirkham @jhamman @lesteve

### CODE REVIEW

File: docs/source/setup/hpc.rst [hunk 1]
```diff
@@ -14,6 +14,34 @@ labs.  These systems commonly have the following attributes:
...
+2.  `dask-drmaa <https://github.com/dask/dask-drmaa>`_ for use with any DRMAA
+    compliant resource manager
+
+They provide interfaces that look like the following:
+
+.. code-block:: python
+
+   from dask_jobqueue import PBSCluster
+
+   cluster = PBSCluster(# TODO realistic example)
```
[the remark this answers is not in the document]
Comment (jhamman) [reply docs/source/setup/hpc.rst]: Here's a real-world example that I've been using on our system (NCAR's Cheyenne):

```Python
cluster = PBSCluster(processes=18, threads=4, memory="6GB",
                     project='P48500028', queue='premium',
                     resource_spec='select=1:ncpus=36:mem=109G',
                     walltime='02:00:00')
```

[the remark this answers is not in the document]
Comment (jakirkham) [reply docs/source/setup/hpc.rst]: There's probably some balance to be struck between a verbatim example and something the average user can grok.

**Comment by jakirkham** [comment]
Generally this seems fine to me once some suitable example is settled on.

File: docs/source/setup/hpc.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/setup/hpc.rst]: Agreed.  However, my experience has been that users familiar with job-queuing systems are actually pretty familiar with all of the terms mentioned in @jhamman 's example.  Are there elements there that you think are particularly confusing and should be removed for simplicity?

**Comment by jhamman** [comment]
I gave this another read through and this all looks good to me.

**Comment by mrocklin** [comment]
Thanks all for the review

File: docs/source/setup/hpc.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/setup/hpc.rst]: I think that it's useful having a concrete example of `resource_spec` there.  *Most* clusters that I've interacted with have needed this explicitly to function well.  For project maybe we replace it with `"..."` ?

**Pull request merged**